modules/RTC/TraceablePeerConnection.js

```diff
@@ -1463,6 +1463,91 @@ TraceablePeerConnection.prototype.removeTrack = function(localTrack) {
     }
 };
 
+/**
+ * Returns the sender corresponding to the given MediaStream.
+ *
+ * @param {MediaStream} stream - The media stream used for the search.
+ * @returns {RTPSender|undefined} - The found sender or undefined if no sender
+ * was found.
+ */
+TraceablePeerConnection.prototype.findSenderByTrack = function(stream) {
+    const track = stream.getTracks()[0];
+
+    if (!track) {
+        logger.error('Cannot find sender: no tracks.');
+
+        return;
+    }
+
+    // Find the right sender (for audio or video)
+    return this.peerconnection.getSenders().find(s => s.track === track);
+};
+
+/**
+ * Replaces <tt>oldTrack</tt> with <tt>newTrack</tt> from the peer connection.
+ * Either <tt>oldTrack</tt> or <tt>newTrack</tt> can be null; replacing a valid
+ * <tt>oldTrack</tt> with a null <tt>newTrack</tt> effectively just removes
+ * <tt>oldTrack</tt>
+ * @param {JitsiLocalTrack|null} oldTrack the current track in use to be
+ * replaced
+ * @param {JitsiLocalTrack|null} newTrack the new track to use
+ * @returns {void}
+ */
+TraceablePeerConnection.prototype.replaceTrack = function(oldTrack, newTrack) {
+    if (browser.isFirefox() && oldTrack && newTrack) {
```

Comment:
1. More work and testing in order to fix all the SSRC mappings for Chrome.
2. I believe sender.replaceTrack will work only on latest Chrome, no idea what is the situation on the other browsers.

It will be great to change the implementation in the following way one day (if possible):
 - during the initialization of the call create fake streams and negotiate.

- For any operations involving changing the streams during the call use this approach without SLD/SRD and renegotiations.

But all this will be a huge task with a lot of changes. I hope soon we'll start working on this but until then it's nice to have the SS working on FF, right!